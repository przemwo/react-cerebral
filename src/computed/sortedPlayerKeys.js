import { compute } from 'cerebral';
import { state } from 'cerebral/tags';

function sortedPlayerKeys (playersKeys, get) {
  return (playersKeys || []).sort((playerAKey, playerBKey) => {
    const playerAScore = get(state`players.data.${playerAKey}.score`);
    const playerBScore = get(state`players.data.${playerBKey}.score`);

    if (playerAScore > playerBScore) {
      return -1;
    } else if (playerAScore < playerBScore) {
      return 1;
    }

    return 0;
  });
}

export default compute(
  state`players.data.*`,
  sortedPlayerKeys
);
