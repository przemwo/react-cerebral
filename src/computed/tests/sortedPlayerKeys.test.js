import { runCompute } from 'cerebral/test';

import sortedPlayerKeys from '../sortedPlayerKeys';

describe('sortedPlayerKeys() computed', () => {
  it('returns sorted player keys', () => {
    const fakeState = {
      players: {
        data: {
          '443fd35f': {
            name: 'player1',
            score: 10
          },
          '187457c2': {
            name: 'player2',
            score: 20
          },
          '5a89a1d9': {
            name: 'player3',
            score: 5
          }
        }
      }
    };
    const result = runCompute(sortedPlayerKeys, { state: fakeState });

    expect(result).toMatchSnapshot();
  });
});
