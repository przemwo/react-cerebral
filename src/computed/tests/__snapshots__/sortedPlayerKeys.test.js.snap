exports[`sortedPlayerKeys() computed returns sorted player keys 1`] = `
Array [
  "187457c2",
  "443fd35f",
  "5a89a1d9",
]
`;
